core/services/workflows/syncer/fetcher.go

```diff
@@ -0,0 +1,59 @@
+package syncer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/validation"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/webapi"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	ghcapabilities "github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/capabilities"
+)
+
+func NewFetcherFunc(
+	ctx context.Context,
+	lggr logger.Logger,
+	och *webapi.OutgoingConnectorHandler,
+	workflowID, workflowExecutionID string,
```

Comment:
You won't have an execution ID when using the syncer

It might be a good idea to have the caller pass in a function that it can use to generate the message ID, or to just have it pass in the ID when calling it

Reply:
got it, will modify it now 🙇🏼